pandas/_libs/lib.pyx

```diff
@@ -2005,7 +2005,8 @@ def maybe_convert_numeric(
     set na_values,
     bint convert_empty=True,
     bint coerce_numeric=False,
-) -> ndarray:
+    bint convert_to_nullable_integer=False,
+) -> "ArrayLike":
```

Comment:
can this be done at a higher level (i.e. not in the cython code)?

Reply:
That would probably require a substantial refactoring of the Python Parser code, which calls this directly. 

FWIW, maybe_convert_objects also has a convert_to_nullable_integer param.